javascript-practice/src/js/services/product.service.js

```diff
@@ -13,4 +13,17 @@ export default class ProductService {
 
     return data;
   }
+
+  /**
+   * Getches a product by its ID
+   * @param {string} id - The ID of the product
+   * @returns {Promise<Object>} The object contains the product
+   */
+  async getProductById(id) {
```

Comment:
No need to use `product` in function because you're define it in product service.

Only use getById(), same for `getProducts`, you can change to getAll()